pgx/src/heap_tuple.rs

```diff
@@ -0,0 +1,340 @@
+//! Provides a safe interface to Postgres `HeapTuple` objects.
+use crate::{
+    heap_getattr_raw, pg_sys, AllocatedByPostgres, AllocatedByRust, FromDatum, FromDatumResult,
+    IntoDatum, PgBox, PgTupleDesc, TriggerTuple, TryFromDatumError, WhoAllocated,
+};
+use std::num::NonZeroUsize;
+
+/// Describes errors that can occur when trying to create a new [PgHeapTuple].
+#[derive(Debug, Copy, Clone)]
+pub enum PgHeapTupleCreationError {
```

Comment:
Anything returned in a `Result<T, E>` as the `E` type should always `impl Error`.  I suggest using `thiserror` which interacts well with `eyre` and `anyhow` to `#[derive(Error)]` this and similar types: https://github.com/dtolnay/thiserror

Reply:
I have no desire to import more dependencies for simple error handling. 

That said I’ll go rtfm on impl Error

Reply:
Done.